dashbuilder-backend/dashbuilder-navigation-backend/src/test/java/org/dashbuilder/navigation/service/LayoutTemplateAnalyzerTest.java

```diff
@@ -161,7 +176,136 @@ public void testIndirectDeadlock() throws Exception {
+        when(navigationServices.loadNavTree()).thenReturn(new NavTreeBuilder()
+                .group("groupC", "", "", true)
+                .item("item1", "", "", true, NavWorkbenchCtx.perspective("A"))
+                .item("item2", "", "", true, NavWorkbenchCtx.perspective("D").setNavGroupId("groupX"))
+                .endGroup()
+                .group("groupX", "", "", true)
+                .item("item3", "", "", true, NavWorkbenchCtx.perspective("D").setNavGroupId("groupX"))
+                .item("item4", "", "", true, NavWorkbenchCtx.perspective("A"))
+                .endGroup()
+                .build());
+
+        LayoutDeadlockInfo info = layoutTemplateAnalyzer.analyzeDeadlock(perspectiveC);
+        assertEquals(info.getRefList().size(), 9);
+        assertEquals(info.getRefList().get(0), new LayoutNavigationRef(PERSPECTIVE, "C"));
+        assertEquals(info.getRefList().get(1), new LayoutNavigationRef(NAV_COMPONENT, TABLIST.getFQClassName()));
+        assertEquals(info.getRefList().get(2), new LayoutNavigationRef(NAV_GROUP_DEFINED, "groupC"));
+        assertEquals(info.getRefList().get(3), new LayoutNavigationRef(DEFAULT_ITEM_DEFINED, "item2"));
+        assertEquals(info.getRefList().get(4), new LayoutNavigationRef(PERSPECTIVE, "D"));
+        assertEquals(info.getRefList().get(5), new LayoutNavigationRef(NAV_COMPONENT, TREE.getFQClassName()));
+        assertEquals(info.getRefList().get(6), new LayoutNavigationRef(NAV_GROUP_CONTEXT, "groupX"));
+        assertEquals(info.getRefList().get(7), new LayoutNavigationRef(DEFAULT_ITEM_FOUND, "item3"));
+        assertEquals(info.getRefList().get(8), new LayoutNavigationRef(PERSPECTIVE, "D"));
+    }
+
+    @Test
+    public void testDefaultItemDeadlock() throws Exception {
+        layoutComponentC1.addProperty(NAV_DEFAULT_ID, "item1");
+        layoutComponentC2.addProperty(NAV_DEFAULT_ID, "item1");
+        when(navigationServices.loadNavTree()).thenReturn(new NavTreeBuilder()
+                .group("groupC", "", "", true)
+                .item("item1", "", "", true, NavWorkbenchCtx.perspective("C"))
+                .item("item2", "", "", true, NavWorkbenchCtx.perspective("B"))
+                .endGroup()
+                .build());
+
+        LayoutDeadlockInfo info = layoutTemplateAnalyzer.analyzeDeadlock(perspectiveC);
+        assertEquals(info.getRefList().size(), 5);
+        assertEquals(info.getRefList().get(0), new LayoutNavigationRef(PERSPECTIVE, "C"));
+        assertEquals(info.getRefList().get(1), new LayoutNavigationRef(NAV_COMPONENT, TABLIST.getFQClassName()));
+        assertEquals(info.getRefList().get(2), new LayoutNavigationRef(NAV_GROUP_DEFINED, "groupC"));
+        assertEquals(info.getRefList().get(3), new LayoutNavigationRef(DEFAULT_ITEM_DEFINED, "item1"));
+        assertEquals(info.getRefList().get(4), new LayoutNavigationRef(PERSPECTIVE, "C"));
+    }
+
+    @Test
+    public void testFirstItemDeadlock() throws Exception {
```

Comment:
This test seems to be the exact duplicate of `testSimpleDeadlock` above.

Reply:
Nice catch! Removed.